ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -25,6 +25,7 @@
 import com.github.ambry.notification.NotificationSystem;
 import com.github.ambry.protocol.RequestOrResponse;
 import com.github.ambry.protocol.RequestOrResponseType;
+import com.github.ambry.store.StoreKey;
```

Comment:
I understand what you are saying. The thing is that the router can only get and fetch ids of type `BlobId`. Any change in ids have to go in as new versions of the `BlobId`. `StoreKey` does not embed any `partition` information and such. `StoreKey` is the abstraction at the server and the `MessageFormat` as they do not try to embed or get the partition information from the id.

Any inconsistency here is all due to the interaction with `MessageFormat`. Anything that is given or received to/from the `MessageFormat` has to be a `StoreKey`. The router also assumes that the chunk ids are `BlobId`s during a get - assuming that chunk ids are `BlobId`s if the metadata blob id is a `BlobId` is a fair assumption I think.

I will look into whether there is anything more we can do to make this usage cleaner and create an issue if needed.